test/rules/creates-lead-salesforce.test.js

```diff
@@ -0,0 +1,83 @@
+'use strict';
+
+const loadRule = require('../utils/load-rule');
+const ContextBuilder = require('../utils/contextBuilder');
+const UserBuilder = require('../utils/userBuilder');
+
+const ruleName = 'creates-lead-salesforce';
+describe(ruleName, () => {
+  let rule;
+  let context;
+  let user;
+  let globals;
+  let stubs = {};
+
+  beforeEach(() => {
+    globals = {
+      request: {
+        post: jest.fn()
+      },
+      auth0: {
+        users: {
+          updateAppMetadata: jest.fn()
+        }
+      },
+      configuration: {
+        SALESFORCE_CLIENT_ID: 'SF_ID',
+        SALESFORCE_CLIENT_SECRET: 'SF_SECRET',
+        SALESFORCE_USERNAME: 'SF_USN',
+        SALESFORCE_PASSWORD: 'SF_PW'
+      }
+    };
+    stubs['slack-notify'] = jest.fn()
+
+    user = new UserBuilder()
+      .build();
+
+    context = new ContextBuilder()
+      .build();
+
+    rule = loadRule(ruleName, globals, stubs);
+  });
+
+  it('should record user as lead and set app metadata', (done) => {
+    const salesforceAccessToken = 'some_token';
+    const salesforceInstanceUrl = 'http://tenant.salesforce.com';
+    
+    rule(user, context, () => { });
+
+    // First POST is to get the access token
+    const getAccessTokenPostOptions = globals.request.post.mock.calls[0][0];
+    expect(getAccessTokenPostOptions.url).toBe('https://login.salesforce.com/services/oauth2/token');
+    expect(getAccessTokenPostOptions.form.grant_type).toBe('password');
+    expect(getAccessTokenPostOptions.form.client_id).toBe(globals.configuration.SALESFORCE_CLIENT_ID);
+    expect(getAccessTokenPostOptions.form.client_secret).toBe(globals.configuration.SALESFORCE_CLIENT_SECRET);
+    expect(getAccessTokenPostOptions.form.username).toBe(globals.configuration.SALESFORCE_USERNAME);
+    expect(getAccessTokenPostOptions.form.password).toBe(globals.configuration.SALESFORCE_PASSWORD);
+    
+    globals.request.post.mock.calls[0][1](null, null,
```

Comment:
referencing `request.post.mock.calls[0][1]` seems strange to me. is the common practice with Jest?